libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -675,144 +693,193 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
     // region QR Code Login Via UI Bridge API Public Implementation
 
     /**
-     * Automatically log in with a UI Bridge API login QR code.
-     * @param loginQrCodeContent The login QR code content.  This should be
-     * either a URL or URL query containing the UI Bridge API JSON parameter.
-     * The UI Bridge API JSON parameter should contain URL-encoded JSON with two
-     * values:
-     * - frontdoor_bridge_url
-     * - pkce_code_verifier
+     * Automatically log in with a UI Bridge API front door bridge URL and PKCE code verifier.
```

Comment:
The remainder of the diff largely reflects my attempt at comprehensively documenting all the properties and methods login activity provides for use in the "reference" implementation of QR code login.  I attempted to name and document everything so we have a clean, understandable and customizable public API.  It's easier to read in Android Studio than git.

@wmathurin - I moved everything back to login activity (from our conversation at DSU) so all apps following our reference implementation's APEX and Visualforce Pages can simply call these reusable methods.  Apps that wish to customize will also have properties they can tweak or they can replace the entire implementation if they're responsible for obtaining the front door URL and code verifier.